src/me/john000708/barrels/Barrel.java

```diff
@@ -42,7 +42,8 @@
     private int capacity;
     private boolean allowDisplayItem;
 
-    public Barrel(Category category, ItemStack item, String name, RecipeType recipeType, final ItemStack[] recipe, int capacity) {
+    // The access can be package-private.
+    Barrel(Category category, ItemStack item, String name, RecipeType recipeType, final ItemStack[] recipe, int capacity) {
```

Comment:
Use protected then. Using the package-private access modifier is not recommended.

Besides, if someone wants to do a subclass for it, they wouldn't have to change it again...